File: ProgettoKalk/KalkMV/Model/poligono.cpp
#include "poligono.h"

Poligono::Poligono(unsigned int n):Figura("Poligono"),nLati(n), p(0){}

double Poligono::distanza(Figura* pf) const{
    double d, d1;
    Segmento* ps = dynamic_cast<Segmento*>(pf);
    if(ps){
        d=Punto::distanza(baricentro(),ps->getA());
        d1=Punto::distanza(baricentro(),ps->getA());
        if(d<d1) return d;
        else return d1;
    }
    Poligono* pp=dynamic_cast<Poligono*>(pf);
    if(pp){
        d=Punto::distanza(baricentro(),pp->baricentro());
        return d;
    }
    Cerchio* pc=dynamic_cast<Cerchio*>(pf);
    if(pc){
        d=Punto::distanza(baricentro(),pc->getCentro());
        return d;
    }
}

File: ProgettoKalk/KalkJava/Poligono.java
public abstract class Poligono extends Figura{
    protected int nLati;
    protected Punto[] p;
    public Poligono(int n){
        super("Poligono");
        nLati=n;

    }

    @Override
    public double distanza(Figura f) {
        double d, d1;
        if (f instanceof Segmento) {
            Segmento s = (Segmento) f;
            d = Punto.distanza(baricentro(), s.getA());
            d1 = Punto.distanza(baricentro(), s.getA());
            if (d < d1) return d;
            else return d1;
        }
        if (f instanceof Poligono) {
            Poligono p = (Poligono) f;
            d = Punto.distanza(baricentro(), p.baricentro());
            return d;
        }
        if (f instanceof Cerchio) {
            Cerchio c = (Cerchio) f;
            d = Punto.distanza(baricentro(), c.getCentro());
            return d;
        }
        return -1;
    }
    public abstract double perimetro();
    public abstract double area();
    public abstract Punto baricentro();

}
